src/ceph-volume/ceph_volume/configuration.py

```diff
@@ -215,3 +225,11 @@ def _read(self, fp, fpname):
             for name, val in options.items():
                 if isinstance(val, list):
                     options[name] = '\n'.join(val)
+
+    def read_conf(self, conffile):
+        if sys_version_info.major >= 3:
```

Comment:
there is no need here to check again for the version, you can check for the attribute instead:

```python
file_reader = getattr(self, 'read_file', None) or getattr(self, 'readfp')
return file_reader(conffile)
```

This allows to avoid checking once again the system version, and does not need to raise a runtime error

Reply:
Although this is more succinct, but this might confuse reader why `read_file` or `readfp` would be absent.

Also, in case of python4, it would raise an `AttributeError` the reason behind which wouldn't be clear. The current code would inform the same saving user the thought/confusion.